source/common/common/logger.cc

```diff
@@ -195,13 +199,15 @@ void Context::activate() {
 }
 
 bool Context::useFineGrainLogger() {
+  absl::MutexLock context_lock(&current_context_mutex);
```

Comment:
we are only reading `current_context` here, so I believe this can be absl::ReaderMutexLock instead.  Same for the few functions below that don't assign `current_context`.